pwndbg/lib/arch.py

```diff
@@ -20,9 +21,10 @@ def update(self, arch_name: str, ptrsize: int, endian: Literal["little", "big"])
         self.ptrmask = (1 << 8 * ptrsize) - 1
         self.endian = endian
 
-        self.fmt = {(4, "little"): "<I", (4, "big"): ">I", (8, "little"): "<Q", (8, "big"): ">Q"}[
-            (self.ptrsize, self.endian)
-        ]
+        if self.endian == "little":
+            self.fmts = {1: "B", 2: "<H", 4: "<I", 8: "<Q"}
+        else:
+            self.fmts = {1: "B", 2: ">H", 4: ">I", 8: ">Q"}
```

Comment:
One nitpick since we are already modifying this, can we move the formatting dicts to global namespace? Otherwise, each `update` call will re-create them (sadly, Python does not optimize such literals (yet?))